test/sql/types/unsigned/test_unsigned_auto_cast.test

```diff
@@ -0,0 +1,103 @@
+# name: test/sql/types/unsigned/test_unsigned_auto_cast.test
+# description: Test unsigned auto cast
```

Comment:
I would also like to see some tests involving storage and checkpointing of unsigned types, see e.g. `test/sql/storage/types/test_hugeint_storage.test`. Also including updates to unsigned columns stored in tables.